styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -793,4 +743,31 @@ private StringValue jsonValue(Object o) throws JsonProcessingException {
         .setExcludeFromIndexes(true)
         .build();
   }
+
+  public <T, E extends Exception> T runInTransaction(TransactionFunction<T, E> f)
+      throws IOException, E {
+    final TransactionalStorage tx = newTransaction();
+    try {
+      final T value = f.apply(tx);
+      tx.commit();
+      return value;
+    } catch (DatastoreException ex) {
+      tx.rollback();
+      throw new TransactionException(false, ex);
+    } finally {
+      if (tx.isActive()) {
+        tx.rollback();
+      }
+    }
+  }
+
+  private TransactionalStorage newTransaction() throws TransactionException {
+    final Transaction transaction;
+    try {
+      transaction = datastore.newTransaction();
+    } catch (DatastoreException e) {
+      throw new TransactionException(false, e);
```

Comment:
Same here - even though paranoid (hard to imagine a `newTransaction` call would "conflict") - if we can easily consistently translate the conflict codes to `true` then we may be saving ourselves some troubleshooting headache later.